torch/nn/parallel/data_parallel.py

```diff
@@ -36,48 +43,71 @@ class DataParallel(Module):
     """
 
     # TODO: update notes/cuda.rst when this class handles 8+ GPUs well
-    def __init__(self, module, device_ids=None, output_device=None):
+
+    def __init__(self, module, device_ids=None, output_device=None, dim=0):
         super(DataParallel, self).__init__()
         if device_ids is None:
             device_ids = list(range(torch.cuda.device_count()))
         if output_device is None:
             output_device = device_ids[0]
+        self.dim = dim
         self.module = module
         self.device_ids = device_ids
         self.output_device = output_device
         if len(self.device_ids) == 1:
             self.module.cuda(device_ids[0])
 
-    def forward(self, *inputs):
+    def forward(self, *inputs, **kwargs):
         def _to_cuda(obj):
             if isinstance(obj, Variable):
                 return obj.cuda()
-            return tuple((map(_to_cuda, obj)))
+            if isinstance(obj, tuple) or isinstance(obj, list):
+                return type(obj)((map(_to_cuda, obj)))
+            return obj
 
         if len(self.device_ids) == 1:
             with torch.cuda.device(self.device_ids[0]):
                 inputs_cuda = _to_cuda(inputs)
-            return self.module(*inputs_cuda)
+                if kwargs:
+                    gpu_dict = {}
+                    for key in kwargs.keys():
+                        gpu_dict[key] = _to_cuda(kwargs[key])
+                    return self.module(*inputs_cuda, **gpu_dict)
+                else:
+                    return self.module(*inputs_cuda)
+
         replicas = self.replicate(self.module, self.device_ids)
         scattered = self.scatter(inputs, self.device_ids)
+
+        gpu_dicts = None
+        if kwargs:
+            scatter_kwargs = {}
+            for key in kwargs.keys():
+                scatter_kwargs[key] = self.scatter(
+                    _to_cuda(kwargs[key]), self.device_ids)
+            gpu_dicts = tuple([
+                dict([(key, values[i])
```

Comment:
`{key: values[i] for key, values in scatter_kwargs.items()}`. Keep it in one line and use `k, v` if necessary.